jvb/src/main/kotlin/org/jitsi/videobridge/EndpointK.kt

```diff
@@ -292,6 +299,83 @@ class EndpointK @JvmOverloads constructor(
         }
     }
 
+    /**
+     * Create an SCTP connection for this Endpoint.  If [Endpoint.OPEN_DATA_LOCALLY] is true,
+     * we will create the data channel locally, otherwise we will wait for the remote side
+     * to open it.
+     */
+    override fun createSctpConnection() {
+        logger.cdebug { "Creating SCTP manager" }
+        // Create the SctpManager and provide it a method for sending SCTP data
+        sctpManager = SctpManager(
+            { data, offset, length ->
+                dtlsTransport.sendDtlsData(data, offset, length)
+                0
+            },
+            logger
+        )
+        sctpHandler.setSctpManager(sctpManager)
+        // NOTE(brian): as far as I know we always act as the 'server' for sctp
+        // connections, but if not we can make which type we use dynamic
+        val socket = sctpManager.createServerSocket()
+        socket.eventHandler = object : SctpSocket.SctpSocketEventHandler {
+            override fun onReady() {
+                logger.info("SCTP connection is ready, creating the Data channel stack")
+                dataChannelStack = DataChannelStack(
+                    { data, sid, ppid -> socket.send(data, true, sid, ppid) },
+                    logger
+                )
+                // This handles if the remote side will be opening the data channel
+                dataChannelStack.onDataChannelStackEvents { dataChannel ->
+                    logger.info("Remote side opened a data channel.")
+                    messageTransport.setDataChannel(dataChannel)
+                }
+                dataChannelHandler.setDataChannelStack(dataChannelStack)
+                if (OPEN_DATA_LOCALLY) {
+                    // This logic is for opening the data channel locally
+                    logger.info("Will open the data channel.")
+                    val dataChannel = dataChannelStack.createDataChannel(
+                        DataChannelProtocolConstants.RELIABLE,
+                        0,
+                        0,
+                        0,
+                        "default"
+                    )
+                    messageTransport.setDataChannel(dataChannel)
```

Comment:
Java scoped this with `Endpoint.this` - we're sure this doesn't pick up a bad definition?

Reply:
Yeah there's no other `messageTransport`